Swiften/MIX/MIXImpl.cpp

```diff
@@ -94,4 +94,95 @@ void MIXImpl::updatePreferences(Form::ref form) {
     request->send();
 }
 
+void MIXImpl::registerNick(const std::string& nick) {
+    auto registerNickPayload = std::make_shared<MIXRegisterNick>();
+    registerNickPayload->setNick(nick);
+    auto request = std::make_shared<GenericRequest<MIXRegisterNick> >(IQ::Set, channelJID_, registerNickPayload, iqRouter_);
+    request->onResponse.connect(boost::bind(&MIXImpl::handleNickAssigned, this, _1, _2));
+    request->send();
+}
+
+void MIXImpl::handleNickAssigned(MIXRegisterNick::ref payload, ErrorPayload::ref error) {
+    std::string nick;
+    if (payload) {
+        nick = payload->getNick();
+    }
+    onNickResponse(nick, error);
+}
+
+void MIXImpl::setNick(const std::string& nick) {
+    auto setNickPayload = std::make_shared<MIXSetNick>();
+    setNickPayload->setNick(nick);
+    auto request = std::make_shared<GenericRequest<MIXSetNick> >(IQ::Set, channelJID_, setNickPayload, iqRouter_);
+    request->onResponse.connect(boost::bind(&MIXImpl::handleNickUpdated, this, _1, _2));
+    request->send();
+}
+
+void MIXImpl::handleNickUpdated(MIXSetNick::ref payload, ErrorPayload::ref error) {
+    std::string nick;
+    if (payload) {
+        nick = payload->getNick();
+    }
+    onNickResponse(nick, error);
+}
+
+void MIXImpl::getSupportedFeatures() {
```

Comment:
The `get*`naming for methods that aren't getters and return things feels strange. Maybe use `request*`instead or so.